src/ContextRegistry.php

```diff
@@ -113,11 +115,11 @@ public function getVariable(string $key, mixed $default = null): mixed
     {
         $context = $this->getCurrentContext();
 
-        if (!isset($context[$key])) {
-            return $default;
+        if (str_starts_with($key, '[')) {
+            return PropertyAccess::createPropertyAccessor()->getValue($context, $key) ?? $default;
```

Comment:
Indeed this is a bit weird. Could we find a way to always use property access?

As a user, I would expect:

```
$context = ['foo' => 'bar', 'nested' => ['a' => 'b']];
$foo = variable('bar');
$a = variable('nested['a']);
```

But When I write this, I really wonder why it's better than

```php
$a = variable('nested')['a']
```

(Note: I'm not against this PR. Just asking)

Reply:
Yes, but the component syntax doesn't allow you to do that, as far as I know. An element of an array must be enclosed in square brackets like `[nested][a]`

The only thing that's better than syntax two (current), is that the default value must be managed again in the case, for example, that the index doesn't exist.

`$a = variable('nested')['a'] ?? null`

And worse if `null` is not the default value

`$a = variable('nested', 'another')['a'] ?? 'another'`

But as indicated in the comment here https://github.com/jolicode/castor/pull/302#issuecomment-1976336797, this approach could be handled in pure PHP without any deps